src/function-builder.ts

```diff
@@ -229,6 +229,19 @@ function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
     }
   }
 
+  if (runtimeOptions.secrets !== undefined) {
+    const invalidSecrets = runtimeOptions.secrets.filter(
+      (s) => !/^[A-Za-z\d\-_]+$/.test(s)
+    );
+    if (invalidSecrets.length > 0) {
+      throw new Error(
+        `Invalid secrets: ${invalidSecrets.join(',')}. ` +
+          'Secret must be configured using the short form (e.g. API_KEY@1) ' +
+          'or with full resource name (e.g. projects/my-project/secrets/API_KEY).'
```

Comment:
Should we update this error message? The tests show that we can only configure secrets in the short form name without versions, but this error message states otherwise.